src/Entity/PageManager.php

```diff
@@ -73,44 +74,7 @@ public function createWithDefaults(array $defaults = []): PageInterface
 
     public function fixUrl(PageInterface $page): void
     {
-        if ($page->isInternal()) {
-            $page->setUrl(null); // internal routes do not have any url ...
-
-            return;
```

Comment:
if you put this in the handlePageurl, you lose the early return

Reply:
Ahhhhh I got it now,  then I will keep internal check here and I will move just hybrid condition

Reply:
Done